File: app/src/main/java/app/beelabs/fingol/ui/adapter/MilestoneListAdapter.java
package app.beelabs.fingol.ui.adapter;

import android.app.Activity;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import app.beelabs.fingol.R;
import butterknife.BindView;
import butterknife.ButterKnife;

import java.util.List;

public class MilestoneListAdapter extends RecyclerView.Adapter<MilestoneListAdapter.ViewHolder> {

    private List<Object> list;
    private Activity activity;

    public MilestoneListAdapter(Activity activity, List<Object> list) {
        this.activity = activity;
        this.list = list;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View layout = LayoutInflater.from(parent.getContext()).inflate(R.layout.row_milestone_task, null, false);


        return new ViewHolder(layout);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        switch (position) {
            case 0:
                holder.date.setText("6");
                holder.month.setText("Jan");
                holder.logo.setImageResource(R.drawable.pie_chart_transfer);
                break;

            case 1:
                holder.datetimeLayout.setVisibility(View.INVISIBLE);
                holder.logo.setImageResource(R.drawable.bpjs_transfer);
                break;

            case 2:
                holder.date.setText("2");
                holder.month.setText("Jan");
                holder.logo.setImageResource(R.drawable.water_electric_transfer);
                break;

            case 3:
                holder.date.setText("1");
                holder.month.setText("Jan");
                holder.title.setText("Top Up");
                holder.title.setTextColor(activity.getResources().getColor(R.color.color_yellow));
                holder.logo.setImageResource(R.drawable.topup_transfer);
                break;

            case 4:
                holder.date.setText("31");
                holder.month.setText("Des");
                holder.title.setText("Payment Done");
                holder.title.setTextColor(activity.getResources().getColor(R.color.color_green));
                holder.logo.setImageResource(R.drawable.water_electric_transfer);
                break;
        }

        holder.detailAction.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                Intent intent = new Intent(activity, JagaCashTransferNoteActivity.class);
//                activity.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return 5;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.detail_action)
        ViewGroup detailAction;
        @BindView(R.id.datetime_layout)
        ViewGroup datetimeLayout;
        @BindView(R.id.date)
        TextView date;
        @BindView(R.id.month)
        TextView month;
        @BindView(R.id.logo)
        ImageView logo;
        @BindView(R.id.title)
        TextView title;

        public ViewHolder(View itemView) {
            super(itemView);

            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/app/beelabs/fingol/ui/adapter/MilestoneListAdapter.kt
package app.beelabs.fingol.ui.adapter

import android.app.Activity
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import app.beelabs.fingol.IConfig
import app.beelabs.fingol.R
import app.beelabs.fingol.ui.activities.TaskDetailActivity
import kotlinx.android.synthetic.main.row_milestone_task.view.*

class MilestoneListAdapter(val activity: Activity, val list: List<Any>?) :
    RecyclerView.Adapter<MilestoneListAdapter.ViewHolder>() {

    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): MilestoneListAdapter.ViewHolder {
        val layout: View = LayoutInflater.from(activity).inflate(R.layout.row_milestone_task, null, false)

        return ViewHolder(layout)
    }

    override fun getItemCount(): Int {
        return 5
    }

    override fun onBindViewHolder(holder: MilestoneListAdapter.ViewHolder, pos: Int) {
        when (pos) {
            0 -> {
                holder.topVertical.visibility = View.INVISIBLE
                holder.date.setText("6");
                holder.month.setText("Jan");
                holder.logo.setImageResource(R.drawable.pie_chart_transfer);
            }

            1 -> {
                holder.datetimeLayout.visibility = View.INVISIBLE
                holder.logo.setImageResource(R.drawable.bpjs_transfer)
            }

            2 -> {
                holder.date.text = "2"
                holder.month.text = "Jan"
                holder.logo.setImageResource(R.drawable.water_electric_transfer)
            }

            3 -> {
                holder.logo.setImageResource(R.drawable.water_electric_transfer)
                holder.date.text = "1"
                holder.month.text = "Jan"
                holder.title.text = "Top Up"
                holder.title.setTextColor(activity.resources.getColor(R.color.color_green))
            }

            4 -> {
                holder.bottomVertical.visibility = View.INVISIBLE
                holder.logo.setImageResource(R.drawable.topup_transfer)
                holder.date.text = "31"
                holder.month.text = "Des"
                holder.title.text = "Payment Done"
                holder.title.setTextColor(activity.resources.getColor(R.color.color_green))
            }

        }

        holder.actionDetail.setOnClickListener(object : View.OnClickListener {
            override fun onClick(p0: View?) {
                val intent = Intent(activity, TaskDetailActivity::class.java)
                intent.putExtra(IConfig.KEY_TITLE_DETAIL, holder.title.text)
                activity.startActivity(intent)
            }
        })
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val topVertical = view.verticalLineTop
        val bottomVertical = view.verticalLineBottom

        val datetimeLayout = view.datetime_layout
        val date = view.date
        val month = view.month
        val logo = view.logo

        val title = view.title
        val content = view.content
        val time = view.time

        val actionDetail = view.detail_action
    }


}